### PR TITLE
Add new flags for Hosted to Astro CLI create and inspect deployment

### PR DESCRIPTION
**Pull request by manmeetkaur**
Relevant slack thread - https://astronomer.slack.com/archives/CL44PA4DB/p1687442314474389

### CODE REVIEW

File: astro/cli/astro-deployment-create.md [hunk 1]
```diff
@@ -37,33 +37,41 @@ Some Deployment configurations, including worker queue and worker type, can be s
...
+| `-d`,`--description` | The description for the Deployment. | Any string. Multiple-word descriptions should be specified in quotations (`"`) |
+| `--enforce-cicd` | Specify that the Deployment can only accept code deploys from API tokens and keys. | None|
+| `-e`,`--executor` | The executor to use for the Deployment. |  CeleryExecutor or KubernetesExecutor |
+| `-a`, `--high-availability` | Enables High Availability for the Deployment. The default is `disable`. | Either `enable` or `disable`. |
+| `-n`,`--name` | The name of the Deployment. | Any string. Multiple-word descriptions should be specified in quotations |
+| `--region` | (Astro Hosted only) The region which you want to use for a Deployment. You will get a prompt to choose the region if a value is not provided. | The [list of regions](resource-reference-hosted?tab=gcp#standard-cluster-configurations) supported for your cloud provider for a standard cluster. |
```
Comment (jwitz) [inline astro/cli/astro-deployment-create.md:52]:
```suggestion
| `--region` | (Astro Hosted only) The region where you want to host the Deployment. | The code for any [supported region](resource-reference-hosted.md#standard-cluster-configurations) |
```

File: astro/cli/astro-deployment-create.md [hunk 2]
```diff
@@ -37,33 +37,41 @@ Some Deployment configurations, including worker queue and worker type, can be s
...
-| `--wait`                    | The time to wait for the new Deployment to have a [healthy](deployment-metrics.md#deployment-health) status before completing the command . | None                                                                                        |
-| `--workspace-id`            | The Workspace in which to create a Deployment. If not specified, your current Workspace is assumed.                                         | Any valid Workspace ID                                                                      |
+| Option | Description | Possible Values |
+| ----- | --- | ------ |
+| `-p`, `--cloud-provider` | The Cloud Provider to use for your Deployment. The default is `gcp` | N/A |
+| `-c`, `--cluster-id` | The cluster in which you want to create a Deployment. You will get a prompt to choose a cluster, if this flag is skipped in case of Astro Hybrid or Astro Hosted Dedicated. | The id of a cluster that you can create Deployments in. See [View clusters](manage-hybrid-clusters#view-clusters) to find your cluster id.|
```
Comment (jwitz) [inline astro/cli/astro-deployment-create.md:43]:
```suggestion
| `-c`, `--cluster-id` | (Astro Hybrid and Astro Hosted dedicated clusters only) The cluster in which you want to create a Deployment | A valid [cluster ID](manage-hybrid-clusters.md#view-clusters).|
```

File: astro/cli/astro-deployment-create.md [hunk 3]
```diff
@@ -37,33 +37,41 @@ Some Deployment configurations, including worker queue and worker type, can be s
...
+| `--cluster-type` | (Astro Hosted only) The type of cluster you want to run the Deployment on. The default is `standard`. | Either `dedicated` or `standard`. |
+| `--dag-deploy` | Enables DAG-only deploys for the Deployment. The default is `disable` | Either `enable` or `disable`. |
+| `--deployment-file` | Location of the template file that specifies the configuration of the new Deployment. File can be in either JSON or YAML format. See [Create a Deployment with a Deployment File](manage-deployments-as-code.md#create-a-deployment-with-a-deployment-file) | A valid file path for YAML or JSON template file |
+| `-d`,`--description` | The description for the Deployment. | Any string. Multiple-word descriptions should be specified in quotations (`"`) |
+| `--enforce-cicd` | Specify that the Deployment can only accept code deploys from API tokens and keys. | None|
+| `-e`,`--executor` | The executor to use for the Deployment. |  CeleryExecutor or KubernetesExecutor |
+| `-a`, `--high-availability` | Enables High Availability for the Deployment. The default is `disable`. | Either `enable` or `disable`. |
```
Comment (jwitz) [inline astro/cli/astro-deployment-create.md:50]:
```suggestion
| `-a`, `--high-availability` | Enables high availibility for the Deployment. The default is `disable`. | Either `enable` or `disable`. |
```

File: astro/cli/astro-deployment-create.md [hunk 4]
```diff
@@ -37,33 +37,41 @@ Some Deployment configurations, including worker queue and worker type, can be s
...
+| `-n`,`--name` | The name of the Deployment. | Any string. Multiple-word descriptions should be specified in quotations |
+| `--region` | (Astro Hosted only) The region which you want to use for a Deployment. You will get a prompt to choose the region if a value is not provided. | The [list of regions](resource-reference-hosted?tab=gcp#standard-cluster-configurations) supported for your cloud provider for a standard cluster. |
+| `-v`,`--runtime-version`    | The Astro Runtime version for the Deployment | Any supported version of Astro Runtime. Major, minor, and patch versions must be specified. |
+| `-s`,`--scheduler-size` | The size of scheduler for the Deployment. The default is `small`. | Possible values can be `small`, `medium` or`large`. |
```
Comment (jwitz) [inline astro/cli/astro-deployment-create.md:54]:
```suggestion
| `-s`,`--scheduler-size` | The size of scheduler for the Deployment. The default is `small`. | Either `small`, `medium`, or `large`. |
```

File: astro/cli/astro-deployment-create.md [hunk 5]
```diff
@@ -37,33 +37,41 @@ Some Deployment configurations, including worker queue and worker type, can be s
...
+| `--wait` | The time to wait for the new Deployment to have a [healthy](deployment-metrics.md#deployment-health) status before completing the command . | None |
```
Comment (jwitz) [inline astro/cli/astro-deployment-create.md:55]:
```suggestion
| `--wait` | Wait for the new Deployment to have a [healthy](deployment-metrics.md#deployment-health) status before completing the command  | None |
```

File: astro/cli/astro-deployment-create.md [hunk 6]
```diff
@@ -37,33 +37,41 @@ Some Deployment configurations, including worker queue and worker type, can be s
...
 
 # Create a Deployment with all required information specified. The CLI will not prompt you for more information
-$ astro deployment create -d="My Deployment Description" --name="My Deployment Name" --cluster-id="ckwqkz36200140ror6axh8p19"
+astro deployment create -d="My Deployment Description" --name="My Deployment Name" --cluster-id="ckwqkz36200140ror6axh8p19"
 
 # Specify the new Deployment's configuration with a yaml file
-$ astro deployment create --deployment-file deployment.yaml
+astro deployment create --deployment-file deployment.yaml
+
+# Create a deployment on Astro Hosted using a Standard cluster
```
Comment (jwitz) [inline astro/cli/astro-deployment-create.md:70]:
```suggestion
# Create a deployment on Astro Hosted using a standard cluster
```

File: astro/cli/astro-deployment-create.md [hunk 7]
```diff
@@ -37,33 +37,41 @@ Some Deployment configurations, including worker queue and worker type, can be s
...
+astro deployment create --name="my-gcp-deployment" --region="us-central1"
+
+# Create a deployment on Astro Hosted using a Dedicated cluster
```
Comment (jwitz) [inline astro/cli/astro-deployment-create.md:73]:
```suggestion
# Create a deployment on Astro Hosted using a dedicated cluster
```

**Review by jwitz — APPROVED** [review]
LGTM! Thanks for handling this so quickly @manmeetkaur 

Remember to check the build errors in the PR to catch any broken links, I see a few relative links that are missing the `.md`

**Comment by manmeetkaur** [comment]
> LGTM! Thanks for handling this so quickly @manmeetkaur
> 
> Remember to check the build errors in the PR to catch any broken links, I see a few relative links that are missing the `.md`

oops! sorry @jwitz will take care in future! :)

**Pull request merged**